src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -276,6 +276,29 @@ protected void CheckEffectiveAccessibility(TypeWithAnnotations returnType, Immut
             diagnostics.Add(Locations[0], useSiteInfo);
         }
 
+        protected void CheckFileTypeUsage(TypeWithAnnotations returnType, ImmutableArray<ParameterSymbol> parameters, BindingDiagnosticBag diagnostics)
```

Comment:
Feels like we should also check type parameter constraints. This also applies to delegates and named types

Reply:
Hmm, saw your comment elsewhere: "I think it's OK to implement a file interface on a non-file type for the same reason it's OK to implement an interface which is less accessible than the implementing type."
If so, then using file-types in constraints would be okay.

Reply:
I noticed that `CheckConstraintTypeVisibility` handles the checks I had in mind. 
I'm not really sure what the benefit of implementing a less-accessible interface then, but it's fine to keep for consistency...